server/supernova/server/main.cpp

```diff
@@ -242,20 +242,19 @@ void set_plugin_paths(server_arguments const & args, nova::sc_ugen_factory * fac
             }
         }
     } else {
-#ifdef __linux__
-        const path home = resolve_home();
-        std::vector<path> folders = { "/usr/local/lib/SuperCollider/plugins",
-                                      "/usr/lib" LIB_SUFFIX "/SuperCollider/plugins",
-                                      home / "/.local/share/SuperCollider/Extensions",
-                                      home / "share/SuperCollider/plugins" };
-
-        for (path const & folder : folders)
-            factory->load_plugin_folder(folder);
-#else
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::Resource) / "plugins");
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::SystemExtension) );
-        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::UserExtension) );
+#ifdef SC_PLUGIN_DIR
+        factory->load_plugin_folder(SC_PLUGIN_DIR);
 #endif
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::Resource) / SC_PLUGIN_DIR_NAME);
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::SystemExtension));
+        factory->load_plugin_folder(SC_Filesystem::instance().getDirectory(DirName::UserExtension));
+        vector<std::string> directories;
+        const char * env_plugin_path = getenv("SC_PLUGIN_PATH");
+        if (env_plugin_path) {
+            for (string const & path : directories) {
```

Comment:
this always iterates over an empty vector; what was intended here?

Reply:
are you trying to iterate over a list of paths in `SC_PLUGIN_PATH`?

Reply:
heauguuauugh bleauaueuauugh